command_line/generate_mask.py

```diff
@@ -32,108 +30,157 @@
-
-    def __init__(self):
-        """ Initialise the script. """
-        from dials.util.options import OptionParser
-        import libtbx.load_env
-
-        # Create the parser
-        usage = "usage: %s [options] experiments.json" % libtbx.env.dispatcher_name
-        self.parser = OptionParser(
-            usage=usage, phil=phil_scope, epilog=help_message, read_experiments=True
-        )
-
-    def run(self):
-        """ Run the script. """
-        from dials.util.masking import MaskGenerator
-        from dials.util.options import flatten_experiments
-        from dials.util import Sorry
-        import six.moves.cPickle as pickle
-        from dials.util import log
-        from dxtbx.format.image import ImageBool
-
-        # Parse the command line arguments
-        params, options = self.parser.parse_args(show_diff_phil=True)
-        experiments = flatten_experiments(params.input.experiments)
-
-        # Configure logging
-        log.config()
-
-        # Check number of args
-        if len(experiments) == 0:
-            self.parser.print_help()
-            return
-
-        if len(experiments) != 1:
-            raise Sorry("exactly 1 experiments must be specified")
-        imagesets = experiments.imagesets()
-        if len(imagesets) > 1:
-            # Check beams (for resolution) and detectors are equivalent in each case
-            # otherwise the mask may not be appropriate across all imagesets
-            detectors = experiments.detectors()
-            beams = experiments.beams()
-            for d in detectors[1:]:
-                if not d.is_similar_to(detectors[0]):
-                    raise Sorry(
-                        "multiple imagesets are present, but their detector"
-                        " models differ"
-                    )
-            for b in beams[1:]:
-                if not b.is_similar_to(beams[0]):
-                    raise Sorry(
-                        "multiple imagesets are present, but their beam"
-                        " models differ"
-                    )
-
-        imageset = imagesets[0]
-
-        # Generate the mask
-        generator = MaskGenerator(params)
-        mask = generator.generate(imageset)
-
-        # Save the mask to file
-        print("Writing mask to %s" % params.output.mask)
-        with open(params.output.mask, "wb") as fh:
-            pickle.dump(mask, fh)
-
-        # Save the experiment list
-        if params.output.experiments is not None:
-            for imageset in imagesets:
-                imageset.external_lookup.mask.data = ImageBool(mask)
-                imageset.external_lookup.mask.filename = params.output.mask
-            from dxtbx.model.experiment_list import ExperimentListDumper
-
-            print("Saving experiments to {0}".format(params.output.experiments))
-            dump = ExperimentListDumper(experiments)
-            dump.as_file(params.output.experiments)
+def script(experiments, params):
```

Comment:
Did we specifically agree that the function should be called `script`? Surely something more descriptive like `generate_mask` would be a lot more intuitive e.g.
```python
from dials.command_line.generate_mask import generate_mask

# later...
generate_mask(exp, params)
"""

Reply:
I agree - don't see the point in having a meaningless class/function name `script` proliferated across the source code.

Reply:
Yep, sorry, I was cargo culting.  I'll give it a proper descriptive name.